src/accountsdb/snapshots.zig

```diff
@@ -47,7 +48,7 @@ pub const Stakes = struct {
     vote_accounts: VoteAccounts,
 
     /// stake_delegations
-    stake_delegations: HashMap(Pubkey, Delegation),
+    stake_delegations: std.AutoArrayHashMap(Pubkey, Delegation),
```

Comment:
Mainly for order determinism. For one it makes it trivial to verify that the exact data which was read from the bincode is written the same way; secondly, we do a lot of iteration over a lot of the maps in the snapshot, which is something that array hash map is highly optimized for. In this specific case, we don't seem to iterate over `stake_delegations` right now anywhere in the code, but the former point still stands, and we may end up needing to iterate over this at some point in the future whenever we start needing to interact with it.

As for array hash map disadvantages: it doesn't proclaim the same speed in lookups, insertion, and deletion - it doesn't state it's very slow though to be clear, and the zig compiler codebase makes heavy use of it.
And since a large amount of our interactions with these hash maps is iteration, we wouldn't strongly benefit from the normal hash map's presumably enhanced lookup/insertion/deletion